.readthedocs.yml

```diff
@@ -16,6 +16,20 @@ build:
   os: ubuntu-22.04
   tools:
     python: "3.11"
+  commands:
+    # Cancel building pull requests when there aren't changes in the docs directory or YAML file.
+    # You can add any other files or directories that you'd like here as well,
+    # like your docs requirements file, or other files that will change your docs build.
+    #
+    # If there are no changes (git diff exits with 0) we force the command to return with 183.
+    # This is a special exit code on Read the Docs that will cancel the build immediately.
+    - |
+      if [ "$READTHEDOCS_VERSION_TYPE" = "external" ] && git diff --quiet origin/main -- docs/ .readthedocs.yaml requirements_docs.txt;
```

Comment:
README and CHANGES (all .rst files) and the source code are part of the documentation.

Reply:
Updated in 9b601824ec2e84a425f00747e3360693a4561213